tools/pipelines-tasks/AVDAppAttachPublish/ReportExceptionTelemetry.ps1

```diff
@@ -0,0 +1,42 @@
+<#
+.SYNOPSIS
+    This script Reports Exceptions to AppAttach Telemetry
+
+.DESCRIPTION
+    This script triggers AppAttachTelemetry DLL with required parameters to capture exceptions in AppAttach Telemetry
+
+.PARAMETER exceptionMessage
+    Specifies the exception message to be captured
+
+.PARAMETER targetDLL
+    Specifies the target DLL to be triggered for AppAttach Telemetry.
+#>
+
+param([string]$exceptionMessage, [string]$targetDLL, [string]$clientType, [string]$clientVersion)
+
+try
+{
+    Write-Output $exceptionMessage
+
+    Add-Type -Path $targetDLL
+
+    # Create and invoke AppAttachTelemetry object.
+    $telemetryProvider = [AppAttachTelemetry.TelemetryProvider]::s_instance
+
+    $appAttachExceptionBuilderObj =  New-Object AppAttachTelemetry.Builders.AppAttachExceptionsBuilder
+    $appAttachExceptionBuilderObj.SetIsSuccessful("False")
+    $appAttachExceptionBuilderObj.SetErrorCode($exceptionMessage)
+    $appAttachExceptionBuilderObj.SetErrorDesc($exceptionMessage)
+    $appAttachExceptionBuilderObj.SetClientType($clientType)
+    $appAttachExceptionBuilderObj.SetClientVersion($clientVersion)
+    $appAttachExceptionBuilderObj.SetState("ErrorOccurred")
+    $appAttachExceptions = $appAttachExceptionBuilderObj.Build()
+
+    $telemetryProvider.ReportException($appAttachExceptions)
```

Comment:
If I'm reading this right, anybody could take your telemetry DLL and use it to send you whatever telemetry they want. I think that would make you unable to trust the data for anything since you cannot be sure it was sent by your code in the way you want.

Reply:
Yes, but this telemetry gets generated only when using it through Azure devops extension as only the release version would have the actual provider id.

Reply:
I see. So I imagine our release pipeline will be fetching a release version of the .nupkg and replacing it when creating the .vsix?

Could you add a comment somewhere that this version of the dll does not send telemetry.

Also, is there any way for customers to prevent the task from submitting telemetry?

Reply:
Yes, release version of app attach framework containing valid provider will be pulled for release version of the vsix
Telemetry will always be generated when using the task. There is no provision to avoid it.